src/components/links/LinkBase.tsx

```diff
@@ -113,9 +122,9 @@ export const LinkBase = forwardRef<HTMLAnchorElement, LinkProps>((props, ref) =>
   }
 
   return (
-    <RouterLink {...restAndRadixProps} onClick={handleClick} ref={ref} to={to}>
+    <RouterNavLink end {...restAndRadixProps} onClick={handleClick} ref={ref} style={style} to={to}>
```

Comment:
Maybe we can add an optional prop to LinkBase to be able to swap the type of link (RouterLink vs RouterNavLink)?
So that we can keep the same behavior and only use NavLink where it's needed?